demisto_sdk/commands/common/GitContentConfig.py

```diff
@@ -12,60 +18,77 @@
-        self.Credentials = GitCredentials()
-        if not repo_name:
+    ENV_REPO_HOSTNAME_NAME = 'DEMISTO_SDK_REPO_HOSTNAME'
+
+    def __init__(self, repo_name: Optional[str] = None, git_provider: Optional[GitProvider] = None, repo_hostname: Optional[str] = None):
+        self.credentials = GitCredentials()
+        self.repo_hostname = repo_hostname or os.getenv(GitContentConfig.ENV_REPO_HOSTNAME_NAME)
+        if not git_provider:
+            git_provider = GitProvider.GitHub
+        self.git_provider = git_provider
+        if not self.repo_hostname:
+            self.repo_hostname = GitContentConfig.GITHUB_USER_CONTENT if git_provider == GitProvider.GitHub else "gitlab.com"
+
+        if 'github.com' in self.repo_hostname:
+            self.repo_hostname = GitContentConfig.GITHUB_USER_CONTENT
+
+        if 'gitlab.com' in self.repo_hostname:
+            self.repo_hostname = 'gitlab.com'
+
+        if not repo_name:  # repo_name is not specified, parsing the remote url the get the details
             try:
-                urls = list(GitUtil().repo.remote().urls)
-                parsed_git = self._get_repository_properties(urls)
+                parsed_git = GitContentConfig._get_repository_properties()
                 self._set_repo_config(parsed_git)
             except (InvalidGitRepositoryError, AttributeError):  # No repository
-                self.CURRENT_REPOSITORY = self.OFFICIAL_CONTENT_REPO_NAME
+                click.secho('No repository was found - defaulting to demisto/content', fg='yellow')
+                self.current_repository = GitContentConfig.OFFICIAL_CONTENT_REPO_NAME
         else:
-            self.CURRENT_REPOSITORY = repo_name
-        if not self.GITLAB_ID:
+            self.current_repository = repo_name
+            self.gitlab_id = self._search_gitlab_id(self.repo_hostname, repo_name)
+
+        if self.git_provider == GitProvider.GitHub:
             # DO NOT USE os.path.join on URLs, it may cause errors
-            self.CONTENT_GITHUB_LINK = urljoin(self.BASE_RAW_GITHUB_LINK, self.CURRENT_REPOSITORY)
-            self.CONTENT_GITHUB_MASTER_LINK = urljoin(self.CONTENT_GITHUB_LINK, r'master')
+            self.base_api = urljoin(GitContentConfig.BASE_RAW_GITHUB_LINK.format(GITHUB_HOST=self.repo_hostname),
+                                    self.current_repository)
+        else:  # gitlab
+            self.base_api = GitContentConfig.BASE_RAW_GITLAB_LINK.format(GITLAB_HOST=self.repo_hostname,
+                                                                         GITLAB_ID=self.gitlab_id)
 
-    def _get_repository_properties(self, urls: Iterable) -> Optional[giturlparse.result.GitUrlParsed]:
+    @staticmethod
+    def _get_repository_properties() -> Optional[giturlparse.result.GitUrlParsed]:
         """Returns the git repository of the cwd.
         if not running in a git repository, will return an empty string
         """
+        urls = list(GitUtil().repo.remote().urls)
```

Comment:
it is an iterator, why we need it to be a list?

Reply:
No real reason, those are leftovers from old code. Will change.